spec/factories/mentors.rb

```diff
@@ -0,0 +1,32 @@
+# == Schema Information
+#
+# Table name: mentors
+#
+#  id         :uuid             not null, primary key
+#  first_name :string           not null
+#  last_name  :string           not null
+#  trn        :string           not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  school_id  :uuid             not null
+#
+# Indexes
+#
+#  index_mentors_on_school_id          (school_id)
+#  index_mentors_on_trn_and_school_id  (trn,school_id) UNIQUE
+#
+# Foreign Keys
+#
+#  fk_rails_...  (school_id => schools.id)
+#
+FactoryBot.define do
+  factory :mentor do
+    association :school
+
+    first_name { Faker::Name.first_name }
+    last_name { Faker::Name.last_name }
+    sequence(:trn) { _1 }
+
+    initialize_with { Mentor.find_or_create_by(first_name:, school:) }
```

Comment:
What's the significance of this? Is it needed?

It'll lead to unusual behaviour because `build(:mentor)` will now perform actions against the database (i.e. find or create) – whereas the 'build' strategy is usually expected to initialise a record but not persist it.

Reply:
It's more to accommodate the seed file, to find or initialize mentors https://github.com/DFE-Digital/itt-mentor-services/blob/452a9e2751aedf0d2340c761250f4f933d1a9111/db/seeds.rb#L51

I'll just not use the factory in the seed file